contentprovider.go

```diff
@@ -552,6 +552,35 @@ func findSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
 	return 0, false
 }
 
+// findMaxOverlappingSection returns the index of the section in secs that
+// overlaps the most with the area defined by off and sz. If no section
+// overlaps, it returns 0, false.
+//
+// The implementation assumes that sections do not overlap.
+func findMaxOverlappingSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
+	// Find the first section that overlaps.
+	j := sort.Search(len(secs), func(i int) bool { return secs[i].End > off })
+
+	if j == len(secs) {
+		return 0, false
+	}
+
+	var maxSz uint32
+	var maxIdx uint32
+	for i := j; i < len(secs); i++ {
+		if secs[i].End <= off || secs[i].Start >= off+sz {
```

Comment:
How is `secs[i].End <= off` possible? `j` should be the first index where `secs[j].End > off`. Then if secs are sorted and non-overlapping then for all i >= j `secs[i].End > off`

The 2nd condition around Start seems like the only needed one?